### PR TITLE
feat(docs): add darkmode

### PR DESCRIPTION
**Pull request by Willem-Jaap**
Nextra has a nice built-in darkmode, this pr enables it along with some small fixes

- feat: enable darkMode
- fix: update svg regex to match actual pattern in imports file
- feat: add logo component with darkmode support, fix import and formatting

### CODE REVIEW

**Comment by Willem-Jaap** [comment]
I upgraded deps, looks like that breaks linting, I'll take a look when I have the time

File: pnpm-lock.yaml
Comment (QuiiBz) [inline pnpm-lock.yaml:1]: Can you revert the dependencies upgrade? In both `docs/package.json` & `pnpm-lock.yaml`

File: docs/pages/index.mdx [hunk 1]
```diff
@@ -17,7 +18,8 @@ import { AppIcon, PagesIcon } from '../components/icons'
 - **Small**: No dependencies, lazy-loaded
 - **Simple**: No Webpack configuration, no CLI, no code generation, just pure TypeScript
 - **Server and Client, Static Rendering**: Lazy-load server and client-side, support for Static Rendering
-- <strong style={{ display: 'inline-flex', gap: '8px', verticalAlign: 'top' }}><AppIcon /> App or <PagesIcon /> Pages Router</strong>: With support for React Server Components
+
```
Comment (QuiiBz) [inline docs/pages/index.mdx:21]: Extra line not needed

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for the PR! Some suggestions:

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**